test/e2e/specs/scaleupdown.go

```diff
@@ -0,0 +1,173 @@
+package specs
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	apiappsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	internalapi "github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/util/managedcluster"
+	"github.com/openshift/openshift-azure/pkg/util/randomstring"
+	"github.com/openshift/openshift-azure/pkg/util/ready"
+	"github.com/openshift/openshift-azure/test/clients/azure"
+	"github.com/openshift/openshift-azure/test/clients/openshift"
+)
+
+var _ = Describe("Scale Up/Down E2E tests [ScaleUpDown][Real][LongRunning][EveryPR]", func() {
+	const (
+		sampleDeployment = "hello-openshift"
+	)
+	var (
+		azurecli          *azure.Client
+		occli             *openshift.Client
+		scaleUpManifest   = flag.String("scaleUpManifest", "../../test/manifests/normal/scaleup.yaml", "Path to the scale up manifest to send in a partial update request to the RP")
+		scaleDownManifest = flag.String("scaleDownManifest", "../../test/manifests/normal/scaledown.yaml", "Path to the scale down manifest to send in a partial update request to the RP")
+		namespace         string
+	)
+
+	BeforeEach(func() {
+		var err error
+		azurecli, err = azure.NewClientFromEnvironment()
+		Expect(err).NotTo(HaveOccurred())
+		occli, err = openshift.NewEndUserClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		namespace, err = randomstring.RandomString("abcdefghijklmnopqrstuvwxyz0123456789", 5)
+		Expect(err).ToNot(HaveOccurred())
+		namespace = "e2e-test-" + namespace
+		fmt.Fprintln(GinkgoWriter, "Using namespace", namespace)
+		err = occli.CreateProject(namespace)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		occli.CoreV1.Namespaces().Delete(namespace, nil)
+	})
+
+	It("should be possible to maintain a healthy cluster after scaling it out and in", func() {
+		ctx := context.Background()
+		ctx = context.WithValue(ctx, internalapi.ContextKeyClientID, os.Getenv("AZURE_CLIENT_ID"))
+		ctx = context.WithValue(ctx, internalapi.ContextKeyClientSecret, os.Getenv("AZURE_CLIENT_SECRET"))
+		ctx = context.WithValue(ctx, internalapi.ContextKeyTenantID, os.Getenv("AZURE_TENANT_ID"))
+
+		By("Fetching the scale up manifest")
+		scaleUp, err := managedcluster.ReadConfig(*scaleUpManifest)
```

Comment:
You shouldn't be reading the external manifest into the internal. You should create a helper that unmarshals the manifest directly into the external struct.